python/activator/middleware_interface.py

```diff
@@ -1262,11 +1262,20 @@ def _check_permanent_changes(self, where: str) -> bool:
         changes : `bool`
             `True` if changes have been made, `False` if retries are safe.
         """
-        data_ids = self.butler.query_data_ids(["instrument", "visit", "detector"], where=where, explain=False)
+        # Need dimension records to determine region.
+        data_ids = self.butler.query_data_ids(
+            ["instrument", "visit", "detector"], where=where, with_dimension_records=True, explain=False
+        )
         if len(data_ids) == 1:
             data_id = data_ids[0]
+            region = data_id.region
+            # Check if processing happened already, limit it by one month.
+            now = astropy.time.Time.now()
+            month = astropy.time.TimeDelta(30, format="jd")
```

Comment:
I think this can be made much smaller, e.g. 1 hour. The point here is to see whether the run that _just failed_ touched the APDB in a way that would be hard to automatically recover.

Reply:
I'll have to change this to use visit time instead of processing time, it won't need an explicit window then.